spec/support/appear_before_matcher.rb

```diff
@@ -0,0 +1,32 @@
+module SortOrderMatchers
```

Comment:
This is cool! Did you get this pattern from somewhere or is it completely hand rolled? I vaguely remember testing sort order can be a pain...